### PR TITLE
tc-proxy: avoid using ServeMux, as it mangles URLs

### LINKED ISSUE #2721: [tc-proxy] Bug when rewriting url with %2F%2F
https://github.com/taskcluster/taskcluster-proxy/issues/44

### PR DESCRIPTION
**Pull request by djmitche**
Github Bug/Issue: Fixes #2721

### CODE REVIEW

File: tools/taskcluster-proxy/routes.go [hunk 1]
```diff
@@ -151,6 +151,7 @@ func (routes *Routes) RootHandler(res http.ResponseWriter, req *http.Request) {
 	defer routes.lock.RUnlock()
 
 	targetPath, err := routes.services.ConvertPath(req.URL)
+	log.Printf("targetPath %s", targetPath)
```
Comment (imbstack) [inline tools/taskcluster-proxy/routes.go:170]: do we want to leave this in here or is it debug cruft? /me happy either way

File: tools/taskcluster-proxy/routes.go [hunk 2]
```diff
@@ -79,6 +79,22 @@ func (routes *Routes) setHeaders(res http.ResponseWriter) {
 	}
 }
 
+// Implement the Handler interface, dispatching requests to the various
+// *Handler methods.  Note that we cannot use ServeMux for this as it mangles
+// URLs and sends redircts.
+func (routes *Routes) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	url := r.URL.String()
+	if strings.HasPrefix(url, "/bewit") {
```
Comment (imbstack) [inline tools/taskcluster-proxy/routes.go:87]: will this handle (and do we want to handle) a situation where this is `"//bewit"`, etc?

**Review by imbstack — APPROVED** [review]

File: tools/taskcluster-proxy/routes.go (hunk 1 above)
Comment (djmitche) [reply tools/taskcluster-proxy/routes.go:170]: It should be removed -- thanks!

File: tools/taskcluster-proxy/routes.go (hunk 2 above)
Comment (djmitche) [reply tools/taskcluster-proxy/routes.go:87]: I don't think we want to handle `http://taskcluster//bewit`, unless I'm misunderstanding the question.

**Pull request merged**